src/hotspot/cpu/riscv/gc/shenandoah/shenandoahBarrierSetAssembler_riscv.cpp

```diff
@@ -64,7 +65,7 @@ void ShenandoahBarrierSetAssembler::arraycopy_prologue(MacroAssembler* masm, Dec
         __ test_bit(t0, t0, ShenandoahHeap::HAS_FORWARDED_BITPOS);
         __ beqz(t0, done);
       } else {
-        __ andi(t0, t0, ShenandoahHeap::HAS_FORWARDED | ShenandoahHeap::MARKING);
```

Comment:
I don't see corresponding changes in the other arches. OTOH, a lot of changes that are in other arches are missing in riscv. Maybe drop the changes and make a separate and complete effort to implement riscv?

Reply:
Yes, it looks the RiscV port is missing quite a lot of the generational mode changes. I'll file a ticket to make the generational mode unavailable on this platform and backout the rest of the RiscV changes.

Reply:
Actually, we do block the generational mode for RiscV. Some of these changes are necessary because of changes to shared symbols. I'll take a closer look.

Reply:
These changes are harmless, but unnecessary after JDK-8314777. Will revert.